app/controllers/projects_controller.rb

```diff
@@ -153,6 +153,22 @@ def archived
     @projects = Project.archived
   end
 
+  def reports
```

Comment:
What do you think of extracting this into its own service class? It seems to me at least, that there is a lot of logic to a controller making harder to maintain later down the road.